Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.cpp

```diff
@@ -291,51 +301,55 @@ ActsExamples::ProcessCode ActsExamples::VertexPerformanceWriter::writeT(
         }
 
         const auto& majorityParticle = *it;
-        associatedTruthParticles.emplace_hint(associatedTruthParticles.end(),
-                                              majorityParticle);
+        trackParameters.push_back(trajectories.trackParameters(tip));
+        associatedTruthParticles.push_back(majorityParticle);
       }
     }
   }
 
-  ACTS_VERBOSE(
-      "Total number of reconstructed tracks : " << trackParameters.size());
-
   // Get number of track-associated true primary vertices
   m_nVtxReconstructable =
-      getNumberOfReconstructableVertices(associatedTruthParticles);
+      getNumberOfReconstructableVertices(SimParticleContainer(
```

Comment:
This makes a copy and then immediately throws it away?

Reply:
Yeah that is the problem with the vector vs set. We use sets for particles but when you do truth matching you want to align your particles with something which isn't straight forward. I changed the internal particle collection to a vector now but still need to convert it if we call helper functions